Source/Project64/N64 System/Mips/Pif Ram.cpp

```diff
@@ -297,19 +297,16 @@ void CPifRam::SI_DMA_READ()
 			{
 				continue;
 			}
-			RDRAM[RdramPos ^3] = m_PifRam[count];
+			RDRAM[BES(RdramPos)] = m_PifRam[count];
 		}
 	}
 	else
 	{
-		for (size_t i = 0; i < 64; i += 4)
-		{
-			unsigned __int32 pif_ram_dword;
-			std::memcpy(&pif_ram_dword, &PifRamPos[i], sizeof(unsigned __int32));
-
-			pif_ram_dword = swap32by8(pif_ram_dword);
+		register size_t i;

```

Comment:
> Yes my method does theoretically expose the variable to unintentional modification.
> However for some………

[Unfortunately the ISO C++ Standard Committee does not agree](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#-p3-express-intent), [nor does CERT (even for C)](https://www.securecoding.cert.org/confluence/display/c/DCL19-C.+Minimize+the+scope+of+variables+and+functions), [nor does Google](https://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Local_Variables), etc, etc. I could go on and on listing examples. If such a practice was more beneficial in the long run, I'm sure your preference would be the one accepted by the standardization committee.

>  Maybe the latter can compile faster

No matter which way you format your code, the difference is negligible. If it wasn't, you wouldn't have a multitude of coding styles for the language (and it would be pretty embarrassing for compiler writers, to be honest).

> Although, this reasoning is rather recursive

You did not ask why, you asked what. If you want the why on that, see [here](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4193.html#809) and [here](https://isocpp.org/files/papers/n4340.html) (the introduction provides your answer in this one, with more detail provided further in).